### ISSUE TITLE
"Cannot join with no overlapping index names" error not appearing in df.corr()

### ISSUE THREAD

**Issue by ramondalmau** [researchpy/documentation#2]
Hi!

I am trying to use the very useful _researchpy_ function _corr_case()_ to compute pairwise p-values for Pearson correlations between delays of flights. 

My input is a Pandas **pivot table** (which I call pivot) with a **single-level Index**, and a **multi-level column**. That is:

pivot.index.names -> FrozenList(['eobd'])
pivot.columns.names -> FrozenList(['adep', 'ades', 'arcid', 'day', 'center_value'])

When I run corr = pivot.corr(), using the standard Pandas functionality, everything is fine. The result is a DataFrame with multi-level index and a multi-level column:

corr.index.names -> FrozenList(['adep', 'ades', 'arcid', 'day', 'center_value'])
corr.columns.names -> FrozenList(['adep', 'ades', 'arcid', 'day', 'center_value'])

where each entry of this DataFrame is the Pearson correlation coeffcient. However, I need the p-value...

When I run rp.correlation.corr_case(pivot) I get the following error:

ValueError                                Traceback (most recent call last)
<ipython-input-65-ca9b9c861875> in <module>
----> 1 corr2 = rp.correlation.corr_case(pivot)
      2 #corr.index.names

/usr/local/lib/python3.6/site-packages/researchpy/correlation.py in corr_case(dataframe, method)
     21 
     22     # Getting r an p value dataframes ready
---> 23     r_vals = dfcols.transpose().join(dfcols, how = 'outer')
     24     p_vals = dfcols.transpose().join(dfcols, how = 'outer')
     25     length = str(len(df))

/usr/local/lib/python3.6/site-packages/pandas/core/frame.py in join(self, other, on, how, lsuffix, rsuffix, sort)
   6813         # For SparseDataFrame's benefit
   6814         return self._join_compat(other, on=on, how=how, lsuffix=lsuffix,
-> 6815                                  rsuffix=rsuffix, sort=sort)
   6816 
   6817     def _join_compat(self, other, on=None, how='left', lsuffix='', rsuffix='',

/usr/local/lib/python3.6/site-packages/pandas/core/frame.py in _join_compat(self, other, on, how, lsuffix, rsuffix, sort)
   6828             return merge(self, other, left_on=on, how=how,
   6829                          left_index=on is None, right_index=True,
-> 6830                          suffixes=(lsuffix, rsuffix), sort=sort)
   6831         else:
   6832             if on is not None:

/usr/local/lib/python3.6/site-packages/pandas/core/reshape/merge.py in merge(left, right, how, on, left_on, right_on, left_index, right_index, sort, suffixes, copy, indicator, validate)
     46                          copy=copy, indicator=indicator,
     47                          validate=validate)
---> 48     return op.get_result()
     49 
     50 

/usr/local/lib/python3.6/site-packages/pandas/core/reshape/merge.py in get_result(self)
    544                 self.left, self.right)
    545 
--> 546         join_index, left_indexer, right_indexer = self._get_join_info()
    547 
    548         ldata, rdata = self.left._data, self.right._data

/usr/local/lib/python3.6/site-packages/pandas/core/reshape/merge.py in _get_join_info(self)
    742             join_index, left_indexer, right_indexer = \
    743                 left_ax.join(right_ax, how=self.how, return_indexers=True,
--> 744                              sort=self.sort)
    745         elif self.right_index and self.how == 'left':
    746             join_index, left_indexer, right_indexer = \

/usr/local/lib/python3.6/site-packages/pandas/core/indexes/base.py in join(self, other, how, level, return_indexers, sort)
   3243             else:
   3244                 return self._join_multi(other, how=how,
-> 3245                                         return_indexers=return_indexers)
   3246 
   3247         # join on the level

/usr/local/lib/python3.6/site-packages/pandas/core/indexes/base.py in _join_multi(self, other, how, return_indexers)
   3341         # need at least 1 in common
   3342         if not overlap:
-> 3343             raise ValueError("cannot join with no overlapping index names")
   3344 
   3345         self_is_mi = isinstance(self, MultiIndex)

ValueError: cannot join with no overlapping index names

Do you know how can I solve this issue? Many thanks in advance!

Ramon

**Comment by researchpy** [comment]
Hey Ramon,

Thanks for the note, are you able to provide sample data for this issue? I was trying to create a pivot table that would replicate what your pivot table's structure for debugging and was able to replicate the error you encountered. Before I started to work on the issue, I tested the Pandas .corr() method and it produced a table, but it only contains values of 1 or -1. Does your table produce decimal values (0.3, 0.6, etc) or are they 1s and -1s?

Thanks and best,

Corey
